socorro/scripts/__init__.py

```diff
@@ -24,3 +24,40 @@ def _fill_text(self, text, width, indent):
                 parts[i] = super(WrappedTextHelpFormatter, self)._fill_text(part, width, indent)
 
         return '\n\n'.join(parts)
+
+
+class FlagAction(argparse.Action):
+    """Facilitates --flag, --no-flag arguments
+
+    Usage::
+
+        parser.add_argument(
+            '--flag', '--no-flag', action=FlagAction, default=True,
+            help='whether to enable flag'
+        )
+
+
+    Which allows you to do::
+
+        $ command
+        $ command --flag
+        $ command --no-flag
+
+
+    And the help works nicely, too::
+
+        $ command --help
+        ...
+        optional arguments:
+           --flag, --no-flag  whether to enable flag
+
+    """
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        super(FlagAction, self).__init__(option_strings, dest, nargs=0, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        if option_string.startswith('--no'):
+            value = False
+        else:
+            value = True
+        setattr(namespace, self.dest, value)
```

Comment:
I really like `--flag/--no-flag` arguments, but Python's argparse makes them kind of hard to do.

We could do `--dump yes/--dump no`, but I like `--flag/--no-flag` better. Amongst other things, we don't have to deal with value typos and we don't have to deal with parsing "yesiness".

We want the two arguments linked because they're two modes of the same thing. I tried using a mutually exclusive group (https://docs.python.org/3/library/argparse.html#mutual-exclusion), but I couldn't get that to work right in regards to the default in the case that you specify neither argument.

Instead of fiddling with that further, I made a `FlagAction` which handles everything. Now we can use it elsewhere.